packages/commons/src/rate-limiter/redisRateLimiter.ts

```diff
@@ -25,21 +26,54 @@ export async function initRedisRateLimiter(config: {
   redisPort: number;
   timeout: number;
 }): Promise<RateLimiter> {
-  const redisClient = await createRedisClient({
+  const redisClient = createRedisClient({
+    // legacyMode: true, // Use legacy mode for compatibility with rate-limiter-flexible
+    disableOfflineQueue: true, // disable the offline queue to make requests fail immediately if Redis is down
+    // so that the rate limiter fallbacks to the insurance limiter
     socket: {
       host: config.redisHost,
       port: config.redisPort,
       connectTimeout: config.timeout,
+      /**
+       * Reconnect indefinitely using a simple exponential back‑off.
+       * `retries` starts at 1 and increases by 1 for every failed attempt.
+       * We cap the delay at 30.000 ms to avoid very long waits.
+       */
+      reconnectStrategy: (retries: number) => Math.min(retries * 1_000, 30_000),
     },
-  })
-    .on("error", (err) => genericLogger.warn(`Redis Client Error: ${err}`))
-    .connect();
+  }).on("error", (err) => genericLogger.warn(`Redis Client Error: ${err}`));
+
+  // Kick‑off the first connection attempt, but do **not** await it.
+  // If Redis is down at start‑up, the promise rejects, we log, and the
+  // client keeps retrying in the background without crashing the service.
+  redisClient
+    .connect()
+    .then(() =>
+      genericLogger.info(
+        "Redis client connected successfully to host " +
+          config.redisHost +
+          ":" +
+          String(config.redisPort)
+      )
+    )
+    .catch((err) =>
+      genericLogger.warn(
+        `Initial Redis connect failed for host ${config.redisHost}:${config.redisPort}, will keep retrying automatically: ${err}`
+      )
```

Comment:
Just make sure you start a rate-limited service and test that these logs are displayed correctly. When I was experimenting with this, it logged an AggregateError instead of what I wrote here (did not investigate further)